Sources/DownloadManager/ViewModel/DownloadManager.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  DownloadManager.swift
+//
+//
+//  Created by Abhishek Pandey on 29/08/23.
+//
+
+import Foundation
+import SwiftUI
+
+public class DownloadManager: ObservableObject {
+    
+    //MARK: - properties
+    var playerManager = PlayerManager()
+    private var downloadManager = DownloadTask()
+    @Published public var taskResult: TaskResult = .progress(0.0)
+    
+    //MARK: Initializer
+    public init() {
+        //configNotification(isRequire: true)
+        downloadManager.downloadAudioCallback = { [self] result in
+            DispatchQueue.main.async {
+                self.taskResult = result
+            }
+        }
+    }
+    
+    //MARK: downloadAudio
+    public func downloadMedia(with url: String, successTitle: String = "", successSubtitle: String = "") {
+        let mediaName = url.components(separatedBy: "/").last ?? ""
```

Comment:
Providing default value of "" to media name in case optional unwrap fails, won't it have any consequences?